transport/inbound.go

```diff
@@ -22,22 +22,41 @@ package transport
 
 //go:generate mockgen -destination=transporttest/inbound.go -package=transporttest go.uber.org/yarpc/transport Inbound
 
-// Inbound is a transport that knows how to receive requests for procedure
-// calls.
+// BaseInbound is the common interface for all inbounds
+type BaseInbound interface {
+	// Stops the inbound. No new requests will be processed.
+	//
+	// This MAY block while the server drains ongoing requests.
+	Stop() error
+
+	// TODO some way for the inbound to expose the host and port it's
+	// listening on
+}
+
+// Inbound is a transport that knows how to receive unary requests for
+// procedure calls.
 type Inbound interface {
+	BaseInbound
+
 	// Starts accepting new requests and dispatches them to the given Handler.
 	//
 	// The function MUST return immediately, although it SHOULD block until
 	// the inbound is ready to start accepting new requests.
 	//
 	// Implementations can assume that this function is called at most once.
 	Start(handler Handler, deps Deps) error
+}
 
-	// Stops the inbound. No new requests will be processed.
-	//
-	// This MAY block while the server drains ongoing requests.
-	Stop() error
+// OnewayInbound is a transport that knows how to receive oneway requests
+// for procedure calls.
+type OnewayInbound interface {
+	BaseInbound
 
-	// TODO some way for the inbound to expose the host and port it's
-	// listening on
+	// Starts accepting new requests and dispatches them to the given Handler.
+	//
+	// The function MUST return immediately, although it SHOULD block until
+	// the inbound is ready to start accepting new requests.
+	//
+	// Implementations can assume that this function is called at most once.
+	Start(handler OnewayHandler, deps Deps) error
```

Comment:
Why have a seperate `OnewayInbound` at all? Why not have a single inbound that gets handed a:

```
type Handlers struct {
  UnaryHandler Handler
  OnewayHandler OnewayHandler
}
```

Then the transport can pull off the wire and choose which one to call...?

@abhinav what do you think?